### PR TITLE
move broker changes to OSXUniversal

### PR DESCRIPTION
**Pull request by kpanwar**
move broker changes to OSXUniversal
<a href='#crh-start'></a><a href='#crh-data-%7B%22approved%22%3A%20%7B%22https%3A//github.com/RPangrle-MS%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%7D%7D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310%23issuecomment-98897908%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22%3Ashipit%3A%22%2C%20%22created_at%22%3A%20%222015-05-05T01%3A11%3A08Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20f906fd83f3140a11e28983fa3382b6ee0f431581%20ADALiOS/ADALiOSTests/ADInstanceDiscoveryTests.m%2015%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310%23discussion_r29638614%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22I%20look%20forward%20to%20the%20day%20that%20someone%20registers%20myfakeauthority.microsoft.com%20for%20something.%22%2C%20%22created_at%22%3A%20%222015-05-05T01%3A09%3A34Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-05-05T18%3A26%3A45Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADALiOS/ADALiOSTests/ADInstanceDiscoveryTests.m%3AL433-443%22%7D%2C%20%22Pull%20f906fd83f3140a11e28983fa3382b6ee0f431581%20ADALiOS/ADALiOS/ADAuthenticationContext.m%20647%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310%23discussion_r29638436%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22self.class%20looks%20REALLY%20weird%20here%2C%20unless%20you%20want%20this%20to%20be%20overridable%20in%20a%20subclass%20I%20would%20use%20ADAuthenticationContext%20%28or%20whatever%20class%20that%27s%20in...%29%20instead.%22%2C%20%22created_at%22%3A%20%222015-05-05T01%3A04%3A04Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/7004783%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/RPangrle-MS%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-05-05T18%3A26%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/5713915%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kpanwar%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20ADALiOS/ADALiOS/ADAuthenticationContext.m%3AL1089-1096%22%7D%7D%2C%20%22processed%22%3A%20%5B%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310%23issuecomment-98897908%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310%23discussion_r29638436%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310%23discussion_r29638614%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310%23discussion_r29697585%22%2C%20%22https%3A//github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310%23discussion_r29697586%22%5D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>

<a href='https://github.com/RPangrle-MS'></a>

- [x] <a href='#crh-comment-Pull f906fd83f3140a11e28983fa3382b6ee0f431581 ADALiOS/ADALiOS/ADAuthenticationContext.m 647'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310#discussion_r29638436'>File: ADALiOS/ADALiOS/ADAuthenticationContext.m:L1089-1096</a></b>
- <a href='https://github.com/RPangrle-MS'></a> self.class looks REALLY weird here, unless you want this to be overridable in a subclass I would use ADAuthenticationContext (or whatever class that's in...) instead.
- [x] <a href='#crh-comment-Pull f906fd83f3140a11e28983fa3382b6ee0f431581 ADALiOS/ADALiOSTests/ADInstanceDiscoveryTests.m 15'></a> &nbsp;<b><a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310#discussion_r29638614'>File: ADALiOS/ADALiOSTests/ADInstanceDiscoveryTests.m:L433-443</a></b>
- <a href='https://github.com/RPangrle-MS'></a> I look forward to the day that someone registers myfakeauthority.microsoft.com for something.

<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/310?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/310?approve=1'></a>&nbsp;<a href='https://www.codereviewhub.com/AzureAD/azure-activedirectory-library-for-objc/pull/310?approve=0'></a>&nbsp;<a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310'></a>
<a href='#crh-end'></a>

### CODE REVIEW

**Comment by brandwe** [comment]
This pull request is now managed with CodeReviewHub. <a href='https://github.com/AzureAD/azure-activedirectory-library-for-objc/pull/310'>Refresh</a>.<a href='#crh-update-none'></a>

File: ADALiOS/ADALiOS/ADAuthenticationContext.m [hunk 1]
```diff
@@ -813,7 +1089,7 @@ -(void) internalAcquireTokenWithResource: (NSString*) resource
         return;
     }
     
-    if (tryCache && promptBehavior != AD_PROMPT_ALWAYS && self.tokenCacheStore)
+    if (![self.class isForcedAuthorization:promptBehavior] && self.tokenCacheStore)
```
Comment (RPangrle-MS) [inline ADALiOS/ADALiOS/ADAuthenticationContext.m]: self.class looks REALLY weird here, unless you want this to be overridable in a subclass I would use ADAuthenticationContext (or whatever class that's in...) instead.

File: ADALiOS/ADALiOSTests/ADInstanceDiscoveryTests.m [hunk 2]
```diff
@@ -431,11 +433,10 @@ -(void) testNonValidatedAuthority
 {
     [self adSetLogTolerance:ADAL_LOG_LEVEL_ERROR];
     NSUUID* correlationId = [NSUUID UUID];
-    [self validateAuthority:@"https://MyFakeAuthority.com/MSOpenTechBV.onmicrosoft.com" correlationId:correlationId line:__LINE__];
+    [self validateAuthority:@"https://MyFakeAuthority.microsoft.com/MSOpenTechBV.onmicrosoft.com" correlationId:correlationId line:__LINE__];
     XCTAssertFalse(mValidated);
```
Comment (RPangrle-MS) [inline ADALiOS/ADALiOSTests/ADInstanceDiscoveryTests.m]: I look forward to the day that someone registers myfakeauthority.microsoft.com for something.

**Comment by RPangrle-MS** [comment]
:shipit:

File: ADALiOS/ADALiOSTests/ADInstanceDiscoveryTests.m (hunk 2 above)
Comment (kpanwar) [inline ADALiOS/ADALiOSTests/ADInstanceDiscoveryTests.m]: :+1:<a href='#crh-update-none'></a>

File: ADALiOS/ADALiOS/ADAuthenticationContext.m (hunk 1 above)
Comment (kpanwar) [inline ADALiOS/ADALiOS/ADAuthenticationContext.m]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**